packages/shared-lib/src/core/model/Studio.ts

```diff
@@ -0,0 +1,51 @@
+// import { BlueprintMapping, TSR } from '@sofie-automation/blueprints-integration'
+
+export interface StudioRouteSetExclusivityGroup {
+	name: string
+}
+
+export interface StudioRouteSet {
+	/** User-presentable name */
+	name: string
+	/** Whether this group is active or not */
+	active: boolean
+	/** Default state of this group */
+	defaultActive?: boolean
+	/** Only one Route can be active at the same time in the exclusivity-group */
+	exclusivityGroup?: string
+	/** If true, should be displayed and toggleable by user */
+	behavior: StudioRouteBehavior
+
+	routes: RouteMapping[]
+}
+export enum StudioRouteBehavior {
+	HIDDEN = 0,
+	TOGGLE = 1,
+	ACTIVATE_ONLY = 2,
+}
+
+export enum StudioRouteType {
+	/** Default */
+	REROUTE = 0,
+	/** Replace all properties with a new mapping */
+	REMAP = 1,
+}
+
+export interface RouteMapping extends ResultingMappingRoute {
+	/** Which original layer to route. If false, a "new" layer will be inserted during routing */
+	mappedLayer: string | undefined
+}
+export interface ResultingMappingRoutes {
+	/** Routes that route existing layers */
+	existing: {
+		[mappedLayer: string]: ResultingMappingRoute[]
+	}
+	/** Routes that create new layers, from nothing */
+	inserted: ResultingMappingRoute[]
+}
+export interface ResultingMappingRoute {
+	outputMappedLayer: string
+	deviceType?: any //TSR.DeviceType
+	remapping?: Partial<any> //BlueprintMapping>
```

Comment:
These missing types are already in shared-lib, so should be easy to link up

Reply:
Yeah, I just had to get some lunch before committing that 🙂